quil-rs/src/instruction/mod.rs

```diff
@@ -319,6 +319,25 @@ impl Quil for Instruction {
     }
 }
 
+pub(crate) struct QuotedString<S>(pub(crate) S);
+
+impl<S> fmt::Display for QuotedString<S>
+where
+    S: AsRef<str>,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "\"")?;
+        for c in self.0.as_ref().chars() {
+            match c {
+                '"' => write!(f, "\\\"")?,
+                '\\' => write!(f, "\\\\")?,
+                c => write!(f, "{}", c)?,
+            }
+        }
```

Comment:
I suspect `String::replace` would be faster than this, but we'd need to benchmark it. I don't know how much the performance of this conversion matters, though.

Reply:
This is specifically inside of `Display`, so I think this might actually be faster. `String::replace` would be creating a new owned `String` and then calling _its_ `Display`, which probably looks similar to this loop.

Reply:
Ah, fair point. 👍🏻